internal/ingest/linksystem.go

```diff
@@ -191,29 +192,72 @@ func (ing *Ingester) ingestAd(publisherID peer.ID, adCid cid.Cid, ad schema.Adve
 		}
 	}
 
-	var maddrs []multiaddr.Multiaddr
-	if len(ad.Addresses) != 0 {
-		maddrs = make([]multiaddr.Multiaddr, 0, len(ad.Addresses))
-		for _, addr := range ad.Addresses {
-			maddr, err := multiaddr.NewMultiaddr(addr)
-			if err != nil {
-				log.Warnw("Bad address in advertisement", "address", addr)
-				continue
-			}
-			maddrs = append(maddrs, maddr)
-		}
-	}
+	maddrs := stringsToMultiaddrs(ad.Addresses)
 
 	provider := peer.AddrInfo{
 		ID:    providerID,
 		Addrs: maddrs,
 	}
-	err = ing.reg.RegisterOrUpdate(context.Background(), provider, adCid, publisher)
+
+	ctx := context.Background()
+
+	err = ing.reg.RegisterOrUpdate(ctx, provider, adCid, publisher)
```

Comment:
I think that there should only be one call to the registry to register the provider as well as any extended providers.  I think it would make sense to add an `extendedProviders` argument to `RegisterOrUpdate`